### ISSUE TITLE
Please add riscv64 FreeBSD support code

### ISSUE THREAD

**Issue by clausecker** [creack/pty#165]
Trying to compile `github.com/ferama/rospo` on riscv64 FreeBSD 13.1.

```
github.com/creack/pty
# github.com/creack/pty
vendor/github.com/creack/pty/pty_freebsd.go:53:20: undefined: fiodgnameArg
vendor/github.com/creack/pty/pty_freebsd.go:66:12: undefined: _C_SPECNAMELEN
vendor/github.com/creack/pty/pty_freebsd.go:69:9: undefined: fiodgnameArg
```

`go version` says `go version go1.20 freebsd/riscv64`

Please add the missing code.

**Comment by adamnyc12** [comment]
github.com/creack/pty
# github.com/creack/pty
vendor/github.com/creack/pty/pty_freebsd.go:53:20: undefined: fiodgnamearg
vendor/github.com/creack/pty/pty_freebsd.go:66:12: undefined: _c_specnamelen
vendor/github.com/creack/pty/pty_freebsd.go:69:9: undefined: fiodgnamearg

**Comment by clausecker** [comment]
Here's the result of running `go tool cgo -godefs types_freebsd.go` on a riscv64-freebsd machine if that helps:

```
// Code generated by cmd/cgo -godefs; DO NOT EDIT.
// cgo -godefs types_freebsd.go

package pty

const (
        _C_SPECNAMELEN = 0xff
)

type fiodgnameArg struct {
        Len     int32
        Buf     *byte
}
```

**Comment by creack** [comment]
Closed by #168

**Comment by clausecker** [comment]
Hm weird, the file you committed has a different value for `_C_SPECNAMELEN`.
